File: Restock_display.py
# Temp solution: should import from Addition Tracker
class AdditionTracker:
    def __init__(self):
        self.additions = []

    def track_addition(self, item, quantity):
        # Track the addition of items
        self.additions.append((item, quantity))

# Temp solution: should import from Expiration Tracker
class ExpirationTracker:
    def __init__(self):
        self.expirations = []

    def track_expiration(self, item, expiration_date):
        # Track the expiration of items
        self.expirations.append((item, expiration_date))

# Temp solution: should import from Slot Tracker
class SlotTracker:
    def __init__(self):
        self.slots = {}

    def track_slot(self, item, slot_number):
        # Track the slot number for items
        self.slots[item] = slot_number
# Temp solution: should import from Purchase Tracker
class PurchaseTracker:
    def __init__(self):
        self.purchases = []

    def track_purchase(self, item, quantity):
        # Track purchases of items
        self.purchases.append((item, quantity))

# Create instances of each tracker
addition_tracker = AdditionTracker()
expiration_tracker = ExpirationTracker()
slot_tracker = SlotTracker()
purchase_tracker = PurchaseTracker()

# Created sample data for the trackers to show that it works
addition_tracker.track_addition("Item A", 10)
expiration_tracker.track_expiration("Item A", "2023-10-31")
slot_tracker.track_slot("Item A", 5)
purchase_tracker.track_purchase("Item A", 5)

# Display relevant tracked information to restockers
def display_information():
    print("Additions:")
    for item, quantity in addition_tracker.additions:
        print(f"{item}: {quantity} added")

    print("\nExpirations:")
    for item, expiration_date in expiration_tracker.expirations:
        print(f"{item}: Expires on {expiration_date}")

    print("\nSlots:")
    for item, slot_number in slot_tracker.slots.items():
        print(f"{item} is in slot {slot_number}")

    print("\nPurchases:")
    for item, quantity in purchase_tracker.purchases:
        print(f"{item}: {quantity} purchased")

# Call the display function
display_information()

File: Restock_display.cpp
#include <iostream>
#include <string>
#include <vector>
#include <unordered_map>

// Addition Tracker class
class AdditionTracker {
public:
    std::vector<std::pair<std::string, int>> additions;

    void track_addition(const std::string& item, int quantity) {
        additions.push_back(std::make_pair(item, quantity));
    }
};

// Expiration Tracker class
class ExpirationTracker {
public:
    std::vector<std::pair<std::string, std::string>> expirations;

    void track_expiration(const std::string& item, const std::string& expiration_date) {
        expirations.push_back(std::make_pair(item, expiration_date));
    }
};

// Slot Tracker class
class SlotTracker {
public:
    std::unordered_map<std::string, int> slots;

    void track_slot(const std::string& item, int slot_number) {
        slots[item] = slot_number;
    }
};

// Purchase Tracker class
class PurchaseTracker {
public:
    std::vector<std::pair<std::string, int>> purchases;

    void track_purchase(const std::string& item, int quantity) {
        purchases.push_back(std::make_pair(item, quantity));
    }
};

int main() {
    // Create instances of each tracker
    AdditionTracker addition_tracker;
    ExpirationTracker expiration_tracker;
    SlotTracker slot_tracker;
    PurchaseTracker purchase_tracker;

    // Created sample data for the trackers to show that it works
    addition_tracker.track_addition("Item A", 10);
    expiration_tracker.track_expiration("Item A", "2023-10-31");
    slot_tracker.track_slot("Item A", 5);
    purchase_tracker.track_purchase("Item A", 5);

    // Display relevant tracked information to restockers
    std::cout << "Additions:" << std::endl;
    for (const auto& addition : addition_tracker.additions) {
        std::cout << addition.first << ": " << addition.second << " added" << std::endl;
    }

    std::cout << "\nExpirations:" << std::endl;
    for (const auto& expiration : expiration_tracker.expirations) {
        std::cout << expiration.first << ": Expires on " << expiration.second << std::endl;
    }

    std::cout << "\nSlots:" << std::endl;
    for (const auto& slot : slot_tracker.slots) {
        std::cout << slot.first << " is in slot " << slot.second << std::endl;
    }

    std::cout << "\nPurchases:" << std::endl;
    for (const auto& purchase : purchase_tracker.purchases) {
        std::cout << purchase.first << ": " << purchase.second << " purchased" << std::endl;
    }

    return 0;
}
